perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_attribute_def_def_o365EmailAddresses_mu.java

```diff
@@ -89,25 +102,33 @@ public void checkAttributeSemantics(PerunSessionImpl sess, Member member, Attrib
 		}
 		String ucoEmail = UCO + "@muni.cz";
 		if (!emails.contains(ucoEmail)) {
-			throw new WrongAttributeValueException(attribute, member, "does not contain " + ucoEmail);
+			throw new WrongReferenceAttributeValueException(attribute, "does not contain " + ucoEmail);
 		}
 
 		//check for duplicities among all members and groups (urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu)
 		AttributesManagerBl attributesManagerBl = sess.getPerunBl().getAttributesManagerBl();
 		Set<Pair<Integer, Integer>> memberPairs = attributesManagerBl.getPerunBeanIdsForUniqueAttributeValue(sess, attribute);
 		memberPairs.remove(new Pair<>(member.getId(), 0));
 		if (!memberPairs.isEmpty()) {
-			Set<Integer> memberIds = new HashSet<>();
-			for (Pair<Integer, Integer> memberPair : memberPairs) {
-				memberIds.add(memberPair.getLeft());
+			try {
+				Member memberWithDuplicateEmail = sess.getPerunBl().getMembersManagerBl().getMemberById(sess, memberPairs.iterator().next().getLeft());
+				throw new WrongReferenceAttributeValueException(attribute, attribute, memberWithDuplicateEmail, null, "some of the email addresses are already assigned.");
+			} catch (MemberNotExistsException e) {
+				throw new ConsistencyErrorException(e);
 			}
-			throw new WrongAttributeValueException(attribute, member, "some of the email addresses are already assigned to the following members: " + memberIds);
 		}
 		Attribute groupO365EmailAddresses = new Attribute(new urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu().getAttributeDefinition());
 		groupO365EmailAddresses.setValue(emails);
 		Set<Pair<Integer, Integer>> groupResourcePairs = attributesManagerBl.getPerunBeanIdsForUniqueAttributeValue(sess, groupO365EmailAddresses);
 		if (!groupResourcePairs.isEmpty()) {
-			throw new WrongAttributeValueException(attribute, "some of the email addresses are already assigned to the following group_resource pairs: " + groupResourcePairs);
+			try {
+				Pair<Integer, Integer> GroupResourceWithDuplicateEmail = groupResourcePairs.iterator().next();
+				Group groupWithDuplicateEmail = sess.getPerunBl().getGroupsManagerBl().getGroupById(sess, GroupResourceWithDuplicateEmail.getLeft());
+				Resource resourceWithDuplicateEmail = sess.getPerunBl().getResourcesManagerBl().getResourceById(sess, GroupResourceWithDuplicateEmail.getRight());
+				throw new WrongReferenceAttributeValueException(attribute, attribute, groupWithDuplicateEmail, resourceWithDuplicateEmail, "some of the email addresses are already assigned." );
```

Comment:
It is attribute group-resource, which means that for this attribute with member (and null in second holder) is there problem with value because of different attribute (not the same, this one is group-resource attribute) and it's holders group and resource.

Reply:
Sorry, I had it wrong, but hopefully it should be correct now.